lib/mail-iso-2022-jp/header.rb

```diff
@@ -3,11 +3,9 @@
 module Mail
   Header.class_eval do
     def encoded
-      buffer = ''
-      fields.each do |field|
+      fields.each_with_object(+'') do |field, buffer|
         buffer << field.encoded rescue Encoding::CompatibilityError
       end
```

Comment:
For a smaller diff, you could change only the line that allocates the string:
```suggestion
      buffer = +''
      fields.each do |field, buffer|
        buffer << field.encoded rescue Encoding::CompatibilityError
      end
      buffer
```